### PR TITLE
Add Domain Cached Credentials 2 (DCC2), MS Cache 2, (NT), -m 31600

### PR DESCRIPTION
**Pull request by Chick3nman**
Add Domain Cached Credentials 2 (DCC2), MS Cache 2, (NT), -m 31600. This is a specialist kernel/module for hash shucking using NT hashes.

Draft PR due to missing test .pm for the module.

### CODE REVIEW

**Comment by Chick3nman** [comment]
Unit test added in PR #3725

File: src/modules/module_31600.c [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+  return tmp_size;
+}
+
+int module_hash_decode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED void *digest_buf, MAYBE_UNUSED salt_t *salt, MAYBE_UNUSED void *esalt_buf, MAYBE_UNUSED void *hook_salt_buf, MAYBE_UNUSED hashinfo_t *hash_info, const char *line_buf, MAYBE_UNUSED const int line_len)
+{
+  u32 *digest = (u32 *) digest_buf;
+
+  hc_token_t token;
+
```
Comment (matrix) [inline src/modules/module_31600.c:71]: @Chick3nman please add here this:

memset (&token, 0, sizeof (hc_token_t));

Comment (Chick3nman) [reply src/modules/module_31600.c:71]: Added

**Comment by Chick3nman** [comment]
Something odd seems to be happening to the two commits I made to add the memsets. Neither is showing up on the PRs, but both are present in their respective branches on my fork. Might just be something delayed in the github UI? Not sure, hopefully they appear and they can be merged without issue. I've converted to draft to see if that changes anything but it doesn't seem like it had any effect.

**Comment by matrix** [comment]
look here https://www.githubstatus.com/

**Comment by Chick3nman** [comment]
Interesting, there appears to have been an "incident with pull requests" but it's marked as resolved and my commits still do not show up in these PRs.

**Comment by Chick3nman** [comment]
Both commits appear on their respective PRs now.

**Pull request merged**